### DISCUSSION TITLE
[Feature Request]: Change out lewisxhe/PCF8563_Library

### DISCUSSION THREAD

**Discussion by ellisgl** [meshtastic/firmware#5038] [Ideas]
### Platform

ESP32

### Description

I see that there has been more active development on one of it's forks: https://github.com/CDFER/pcf8563-RTC

**Comment by ellisgl** [comment]
Assign it to me - I'll get it done. =)

**Comment by ellisgl** [comment]
https://github.com/meshtastic/firmware/pull/5031

**Comment by caveman99** [comment]
Could you explain what is the difference? E.G. what actual problem does this solve? We generally stick with the upstream libs and advocate merging such changes upstream 'it is more active in development' can mean everything and nothing.

**Comment by ellisgl** [comment]
@caveman99 

- lewisxhe/PCF8563_Library was last updated 2022-10-06.
- CDFER/pcf8563-RTC was created in 2023-03-21, and was last updated 2024-08-17.
- CDFER/pcf8563-RTC name of the lib actually tells you what's it's for.
- CDFER/pcf8563-RTC is way better documented. lewisxhe/PCF8563_Library has a name and a simple example.
- CDFER/pcf8563-RTC has several bug fixes:
    - [Fix sync to RTC]( https://github.com/CDFER/pcf8563-RTC/commit/3fcc8b16a6639625eae13d4c62ec717de7ebc1ad)
    - [Timezone Bugfix](https://github.com/CDFER/pcf8563-RTC/commit/4094ad64645ecaa2a65e4197de64f9de717abf39)
    - [Make sync functions fail safe](https://github.com/CDFER/pcf8563-RTC/commit/945b4fe36b6670f7bade740d5aeabb8ea0309d3c)
- CDFER/pcf8563-RTC has several improvements:
    - [Use UTC time not local timezone](https://github.com/CDFER/pcf8563-RTC/commit/667f5c3cb5caa09029d485e0c3ec0faff42afab2)
    - [Faster epoch creation](https://github.com/CDFER/pcf8563-RTC/commit/582c7da69e5419648cef3e0c5266e96d8b042c40)
    - [Add error to text function](https://github.com/CDFER/pcf8563-RTC/commit/9f54feb3e3ab7951837d811ec5288453194741d0)

**Comment by caveman99** [comment]
From the library readme:

- Not all functions are implemented
- Not compatible with Lewis' library
- Only tested with the ESP32 and S2 variants

We use this with all kinds of processor arches, and the way you built up the code request you assume it's a drop in replacement, which it is not. Alone the change from local zone to UTC requires code changes in meshtastic. I am not in favor of doing this.

**Comment by ellisgl** [comment]
@caveman99 I thought I saw it said something about having backwards compatibility. I'm feeling like a stooge now. I think I'll dig more into this and see what I can do about solving those issues.

**Reply by fifieldt** [reply]
Appreciate your work @ellisgl !
